spec/std/spec/filters_spec.cr

```diff
@@ -0,0 +1,141 @@
+require "spec"
+
+module Spec::Item
+  setter focus : Bool
+end
+
+private class FakeRootContext < Spec::RootContext
+  def description
+    "root"
+  end
+end
+
+private def all_descriptions(item) : Array(String)
+  child_descriptions = item.responds_to?(:children) ? item.children.flat_map { |c| all_descriptions(c) } : [] of String
+  child_descriptions.unshift(item.description)
+end
+
+private def build_spec(filename, root = nil, count = 2)
+  root ||= FakeRootContext.new
+  name = filename.chomp(".cr")
+
+  1.upto(count) do |i|
```

Comment:
@straight-shoota I left this as original because `1.upto(count)` gives `1, 2, 3` whereas `count.times` gives `0, 1, 2`.  Aside from the lines, the value is also used as part of the context and example naming, and that ends up switching them all to 0 based, which I found harder to read.